api/doc/domain-wide-delegation.md

```diff
@@ -0,0 +1,48 @@
+# Domain-wide delegation in AoU RW
```

Comment:
I'm trying a new thing here, adding some relatively-lightweight .md documentation into the codebase. Compared to a separate tech doc, this makes it more easily linkable and searchable within the repo.

I'm hoping this might be a good pattern for documenting "the way things work" for certain parts of the system that may be non-obvious. This was a pretty successful pattern at Google (including team-focused documentation inside the codebase); we'll see if it catches on here :)